### PR TITLE
Git note update

### PR DESCRIPTION
**Pull request by odysseus0**
Added some additional notes to the git-notes, based on the bugs I run into when following the notes step by step.

### CODE REVIEW

File: docs/dev/git.md [hunk 1]
```diff
@@ -189,11 +189,21 @@ It's very important that you **always work inside a branch**. If you make any co
    git clone https://github.com/fastai/fastai
    cd fastai
    ```
+
+   Uninstall any previously installed `fastai` library:
+
+   ```bash
+   pip uninstall fastai
+   conda uninstall fastai
```
Comment (odysseus0) [inline docs/dev/git.md]: The guide forgot to remind people to uninstall potentially installed fastai library.

Comment (stas00) [reply docs/dev/git.md]: Yes!
Though only conda needs to be uninstalled. pip will uninstall its preinstalled package when installing an editable one. See: https://docs.fast.ai/dev/develop.html#editable-install-explained

File: docs/dev/git.md [hunk 2]
```diff
@@ -189,11 +189,21 @@ It's very important that you **always work inside a branch**. If you make any co
...
+   ```
+
    and install the dev prerequisites:
    ```
    pip install -e .[dev]
    ```
 
+   *Note: if you are using zsh as your default shell, the last command will not work for you. Simply execute `bash` before running it.*
```
[the remark this answers is not in the document]
Comment (stas00) [reply docs/dev/git.md]: I'd be happy to add this note, for sure, but would you please investigate why this is a problem in first place? There are no such notes in `pip` documentation.

Does it work with:

pip install -e .
or
pip install -e ".[dev]"
?

File: docs/dev/git.md (hunk 1 above)
Comment (stas00) [reply docs/dev/git.md]: I added a whole sub-step explaining that.

**Comment by stas00** [comment]
Thank you for the suggestions, @odysseus0 - I have applied them all with extra prose and tweaks. If anything else needs to be done, please let me know.

**Comment by odysseus0** [comment]
Thank you again for writing the detailed and thorough guide! Sorry that I could not reply in a timely fashion due to the time zone difference.

File: docs/dev/git.md (hunk 1 above)
Comment (odysseus0) [reply docs/dev/git.md]: Thanks for the explanation!

**Pull request closed without merging**